### DISCUSSION TITLE
class values and data-* attributes removed

### DISCUSSION THREAD

**Discussion by bart-d** [requarks/wiki#6120] [Help / Questions]
Hi,

I'm including javascript through the theme settings which modifies certain links to show a tooltip and modify their color. The tooltip works, but the changes to the appearance of the link isn't. Even when i literaly put a link HTML mode like <a href="#" class="abc" data-abc="xyz"> in the page, the class and data- attribute is removed when being published. 

How can i work around this? I've tried deferring the loading of the JS and disabling most if not all of the rendering and security options, but no luck. Sometimes it briefly works when i haven't visited the page in a while, so that might be cache related, but after a refresh i'm back to the issue. Any feedback or pointers are appreciated.

**Comment by NGPixel** [comment]
Have you disabled HTML Sanitization under Admin > Rendering > HTML > Security?

**Comment by bart-d** [comment]
Yes, this is indeed one of the options i tried to disable. In fact every single option onder Rendering > HTML is currently disabled.

**Comment by Tuege** [comment]
@bart-d Did you manage to solve this issue? I am currently trying to implement tooltips too and am running into the same issue where the html doesn't want to play nice with the js script. I have already tried including the script directly in the html with no luck and disabling the html cleaning option.

The specific command that doesn't work for me is
`const tooltipContainer = document.querySelector(".tooltip-container");` which returns `null`

ChatGPT seems to think this has something to do with the order of loading the html and js script as you mentioned above:
`It's also possible that the element you are trying to access is not yet loaded when your JavaScript code is executed. You can try moving your JavaScript code to the bottom of the HTML file, just before the closing </body> tag, to ensure that all elements in the DOM are loaded before the code runs. Alternatively, you can wrap your JavaScript code in a DOMContentLoaded event listener to make sure it runs only after the DOM has finished loading`

**Comment by Tuege** [comment]
ok I have just found the solution in #5573. If you look at NGPixel's last reply that works for me (At least as far I can initially tell)
